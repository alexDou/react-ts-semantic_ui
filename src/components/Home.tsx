import React, { ReactElement } from 'react';
import {
    Segment, Container, Header, Grid, Image, Icon
} from 'semantic-ui-react';
import { style } from 'typestyle';

import SearchContainer from '@containers/search';

const styles = {
    homeSegment: style({
        $nest: {
            '&.ui.vertical.segment': {
                margin: '5em 0em 0em',
                padding: '5em 0em',
            },
        }
    }),
    homeContainer: style({
        marginTop: '2em',
    }),
    homeGrid: style({
        $nest: {
            '&.ui.grid': {
                maxWidth: 450,
                margin: '0 auto'
            }
        }
    })
};

const Home = (): ReactElement => {
    return (
        <Segment vertical className={styles.homeSegment}>
            <Container textAlign="center">
                <Header as="h1">
                    <Image src="assets/images/logo.png" />
                </Header>
                <Container text className={styles.homeContainer}>
                    <Grid textAlign="center" className={styles.homeGrid} verticalAlign="middle">
                        <Grid.Column>
                            <Header as="h2" color="black" textAlign="center">
                                <Icon name="github square" />
                                Github Repositories
                            </Header>
                            <SearchContainer
                                size="large"
                                color="black"
                                stacked={true}
                                disabled={false}
                                query=""
                            />
                        </Grid.Column>
                    </Grid>
                </Container>
            </Container>
        </Segment>
    );
}

export default Home;
