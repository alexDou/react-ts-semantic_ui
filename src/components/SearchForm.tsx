import React, { useState } from 'react';
import { Button, Form, Segment } from 'semantic-ui-react';
import { style } from 'typestyle';

import { SearchFormProps } from '@t/app';

const inputStyle = style({
    minWidth: '260px'
})

const SearchForm: React.FC<SearchFormProps> = (props) => {
    const { size, stacked, color, disabled, query, onFormSubmit } = props;
    const [q, setQ] = useState(query);

    const handleChange = (e: any) => setQ(e.target.value);
    const handleSubmit = () => onFormSubmit(q as string);

    const formStyle = style({
        $nest: {
            '&.ui.form': {
                maxWidth: size === 'small' ? '400px' : '100%',
                margin: size === 'small' ? '0 auto' : 'initial'
            }
        }
    });

    const form = () =>
        <>
            <Form.Input
                disabled={disabled}
                fluid
                icon="search"
                iconPosition="left"
                placeholder="discover best ideas"
                type="text"
                value={q}
                className={inputStyle}
                onChange={handleChange}
            />
            <Button disabled={disabled} basic color={color} fluid size={size}>
            Search
            </Button>
        </>

    return (
        <Form size={size} onSubmit={handleSubmit} className={formStyle}>
            <Segment stacked={!!stacked}>
                {size === 'small'
                    ? <Form.Group>{form()}</Form.Group>
                    : form()
                }
            </Segment>
        </Form>
    )
}

export default SearchForm;
