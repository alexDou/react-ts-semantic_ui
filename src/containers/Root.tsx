import React from 'react';
import { Provider } from 'react-redux';

import { RootProps } from '@t/app';
import store from '@store/store';

const Root = ({ iniStore, children }: RootProps) => {
    return (
        <Provider store={store(iniStore)}>
            {children}
        </Provider>
    );
};

export default Root;
