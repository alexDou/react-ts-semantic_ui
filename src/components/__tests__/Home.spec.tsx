import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import Root from '@containers/Root';
import Home from '@components/Home';
import SearchForm from '@components/SearchForm';
import SearchContainer from '@containers/search';
import {
    Segment, Container, Header
} from 'semantic-ui-react';

describe('Home component', function() {
    it('renders itself', () => {
        const home = shallow(
            <Home />
        );

        expect(home.find(Segment)).toHaveLength(1);
    });
    it('renders UI components', () => {
        const home = mount(
            <Root iniStore={{} as any}>
                <BrowserRouter>
                    <Home />
                </BrowserRouter>
            </Root>
        );

        expect(home.find(Container)).toHaveLength(2);
        expect(home.find(Header)).toHaveLength(2);

        home.unmount();
    });
    it('renders SearchContainer and SearchForm', () => {
        const home = mount(
            <Root iniStore={{} as any}>
                <BrowserRouter>
                    <Home />
                </BrowserRouter>
            </Root>
        );

        expect(home.find(SearchContainer)).toHaveLength(1);
        expect(home.find(SearchForm)).toHaveLength(1);
        home.unmount();
    });
    it('matches snapshot', () => {
        const home = mount(
            <Root iniStore={{} as any}>
                <BrowserRouter>
                    <Home />
                </BrowserRouter>
            </Root>
        );

        expect(home).toMatchSnapshot();
        home.unmount();
    });
});
