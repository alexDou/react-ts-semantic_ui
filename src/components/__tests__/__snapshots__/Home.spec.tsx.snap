// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `ReactWrapper {}`;
